From 5ff9d20dfa71c124f1a14c1e3cbb776aa7af46c2 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Fri, 10 May 2024 14:34:59 +0000
Subject: [PATCH] FmpDevicePkg: Force accept old UEFI capsules updates

In the case of a rollback from 35.5.0 to 35.4.1, we need to apply the
old capsule. The version check in the UEFI firmware does not allow for
that, so we need to disable the check to perform a clean and reliable
rollback.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 FmpDevicePkg/FmpDxe/FmpDxe.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/FmpDevicePkg/FmpDxe/FmpDxe.c b/FmpDevicePkg/FmpDxe/FmpDxe.c
index 1e7ec4a09e..bf42c8e706 100644
--- a/FmpDevicePkg/FmpDxe/FmpDxe.c
+++ b/FmpDevicePkg/FmpDxe/FmpDxe.c
@@ -982,13 +982,15 @@ CheckTheImageInternal (
   if (Version < Private->Descriptor.LowestSupportedImageVersion) {
     DEBUG (
       (DEBUG_ERROR,
-       "FmpDxe(%s): CheckTheImage() - Version Lower than lowest supported version. 0x%08X < 0x%08X\n",
+       "FmpDxe(%s): CheckTheImage() - Version Lower than lowest supported version. 0x%08X < 0x%08X - continuing anyway\n",
        mImageIdName, Version, Private->Descriptor.LowestSupportedImageVersion)
       );
+    /*
     *ImageUpdatable    = IMAGE_UPDATABLE_INVALID_OLD;
     Status             = EFI_SUCCESS;
     *LastAttemptStatus = LAST_ATTEMPT_STATUS_DRIVER_ERROR_VERSION_TOO_LOW;
     goto cleanup;
+    */
   }
 
   //
-- 
2.17.1

